framework/HypermediaResult.js

```diff
@@ -0,0 +1,145 @@
+import { componentStoreFactory, isPseudoTag } from '../render/componentFactory.js';
+import { defaultTemplateProcessor, TemplateResult } from 'lit-html';
+import { fetch, stateFactory } from '../state/store.js';
+import { html } from '../framework/hypermedia-components.js';
+import { observableTypes } from '../state/HypermediaState.js';
+import { until } from 'lit-html/directives/until.js';
+
+export function custom(tag, elementClass, pseudoTag, hypermediaClasses, options) {
+	pseudoTag = pseudoTag || tag;
+	const components = componentStoreFactory(pseudoTag);
+	components.register(tag, hypermediaClasses);
+	customElements.define(tag, elementClass, options);
+}
+
+/**
+ * Extension to the TemplateResult class from lit-html
+ * Contains the logic for replacing pseudotags with tags
+ */
+export class HypermediaResult extends TemplateResult {
+
+	constructor(strings, values, type, processor) {
+		super(strings, values, type, processor);
+		this.processing();
+	}
+
+	/*
+	* Function called on HypermediaResult object to get the associated html
+	* Should process the psuedocode if this is the first call to getHTML
```

Comment:
```suggestion
	* Should process the base tag if this is the first call to getHTML
```
I think you meant this?